### PR TITLE
CVar CharacterRequirement

### PR DESCRIPTION
**Pull request by sleepyyapril**
# Description

This takes in one CVar and one required value and simply checks if the CVar's value as a string is equal to the required value. I could make it use a list of them, but I really didn't think it was needed considering CharacterAndLogic.

I also cleaned up CharacterRequirements.Profile.cs a bit.

### CODE REVIEW

File: Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+using Content.Shared.Roles;
+using JetBrains.Annotations;
+using Robust.Shared.Configuration;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Customization.Systems;
+
+/// <summary>
+///     Requires the profile to have one of the specified loadouts
```
Comment (FoxxoTrystan) [inline Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs:12]: not sure if this is the right summary

Comment (sleepyyapril) [reply Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs:12]: good catch

File: Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace Content.Shared.Customization.Systems;
+
+/// <summary>
+///     Requires the server to have a specific CVar value. 
+/// </summary>
+[UsedImplicitly, Serializable, NetSerializable,]
+public sealed partial class CVarRequirement : CharacterRequirement
+{
+    [DataField("cvar", required: true)]
```
Comment (VMSolidus) [inline Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs:17]:
```suggestion
    [DataField(required: true)]
```

File: Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    public string CVar;
+
+    [DataField("requiredValue", required: true)]
```
Comment (VMSolidus) [inline Content.Shared/Customization/Systems/CharacterRequirements.Misc.cs:20]:
```suggestion
    [DataField(required: true)]
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

File: Resources/Locale/en-US/customization/character-requirements.ftl [hunk 4]
```diff
@@ -140,3 +140,11 @@ character-whitelist-requirement = You must{$inverted ->
     [true]{" "}not
     *[other]{""}
 } be whitelisted
+
+## Cvar
```
Comment (DEATHB4DEFEAT) [inline Resources/Locale/en-US/customization/character-requirements.ftl:144]:
```suggestion
## CVar
```

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Pull request merged**